### PR TITLE
enable indexing with GoLand in prebuilds

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

This PR enables indexing with GoLand in prebuilds.

## Related Issue(s)

Fixes #

## How to test

- Enable GoLand in settings: gitpod.io/preferences
- Start a workspace on this branch and check that indexing is already done: https://gitpod.io#https://github.com/gitpod-io/gitpod/pull/9193

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@akosyakov: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: .gitpod.yml [hunk 1]
```diff
@@ -47,6 +47,21 @@ tasks:
     before: pre-commit install --install-hooks
     init: leeway exec --filter-type go -v -- go mod verify
     openMode: split-right
+  - name: GoLand indexing
+    init: |
+      if [ ! "$GITPOD_HEADLESS" = "true" ] ; then exit ; fi
```
Comment (easyCZ) [inline .gitpod.yml:52]: Given this is quite a heavy-weight script, does it make sense to check it in as a script and reference in `.gitpod.yml`?

Comment (akosyakov) [reply .gitpod.yml:52]: sure, let's try fist whether we can make it actually work 😄

**Comment by akosyakov** [comment]
It seems [second prebuild](https://gitpod.io/t/gitpod/gitpod/adcf7027-1d33-48d3-8477-2cb773a865a3) stuck. @gitpod-io/engineering-workspace maybe you can investigate how can we improve it? 🙏

File: scripts/goland-indexing.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+curl -sSLo backend.tar.gz "$JB_BACKEND_DOWNLOAD_URL" && tar -xf backend.tar.gz --strip-components=1 && rm backend.tar.gz
+
+# config JB system config and caches aligned with runtime
+printf '\nshared.indexes.download.auto.consent=true' >> "/tmp/backend/bin/idea.properties"
+unset JAVA_TOOL_OPTIONS
+export IJ_HOST_CONFIG_BASE_DIR=/workspace/.config/JetBrains
+export IJ_HOST_SYSTEM_BASE_DIR=/workspace/.cache/JetBrains
+
+# start JB backend in indexing mode
+/tmp/backend/bin/remote-dev-server.sh warmup "$GITPOD_REPO_ROOT"
```
Comment (princerachit) [inline scripts/goland-indexing.sh:24]: nit: newline

**Comment by princerachit** [comment]
I am going to retrigger the build and investigate.

**Comment by princerachit** [comment]
> I am going to retrigger the build and investigate.

It just worked 😆 without and issue.

**Comment by akosyakov** [comment]
That's the problem: It is not reliable. Sometimes it is good, sometimes it hangs. Maybe depends on which node it ends up and load. Maybe try to run several times.

**Comment by akosyakov** [comment]
If everybody happy though for now, we can merge and see whether it will cause issues for us on main.

**Review by easyCZ — DISMISSED** [review]

**Comment by easyCZ** [comment]
/hold

[nit] Newline in the bash

**Comment by akosyakov** [comment]
@easyCZ addressed a nit, unfortunately you need to approve again 😬

**Review by easyCZ — APPROVED** [review]

**Pull request merged**